hw/ip/spi_device/lint/spi_device.waiver

```diff
@@ -164,3 +167,6 @@ waive -rules {ASSIGN_SIGN NEG_ASSIGN} -location {spi_passthrough.sv} \
 waive -rules {LATCH COMBO_NBA} -location {spi_readcmd.sv} \
       -regexp {'last_read_address_o'} \
       -comment "Intentional use of latch. And guideline is to use non-blocking assignment"
+waive -rules {IFDEF_CODE} -location {spi_readcmd.sv} \
```

Comment:
QQ regarding the use of a latch below: 
I see that there is also a flops-based implementation for the FPGA case. 
Is there a specific reason why we are not using that flop-based implementation all the time? 
We could get rid of the latch and associated issues with timing constraints if we did so...

Reply:
I am not sure CSb can be a clock in general. I am more of worrying the  low slew rate of CSb signal that may cause issues on the logic. Didn't check the CSb signal requirement on the SPI spec though. Probably It won't be in the spec.

Reply:
![SPI Device Design Change_Page_11](https://user-images.githubusercontent.com/1192814/127574270-4095da3d-a4a0-4c47-90f0-bef8fc57d682.png)

Reply:
@tjaychen commented in #7415 . I think this is safer than using CSb as a clock. I've implemented this scheme in #7415

Reply:
Ah yes I see. Yes this scheme looks safer to me, thanks for changing that!

Sorry for commenting on that here, I realize now that this particular commit belongs to another PR.